x-pack/plugins/canvas/public/expression_types/datasource.js

```diff
@@ -6,14 +6,20 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import { FormattedMessage } from '@kbn/i18n/react';
 import { Registry } from '../../common/lib/registry';
 import { RenderToDom } from '../components/render_to_dom';
 import { ExpressionFormHandlers } from '../../common/lib/expression_form_handlers';
 import { BaseForm } from './base_form';
 
 const defaultTemplate = () => (
   <div>
-    <p>This datasource has no interface. Use the expression editor to make changes.</p>
+    <p>
+      <FormattedMessage
+        id="xpack.canvas.expressionTypes.dataSourceHasNoInterfaceErrorMessage"
```

Comment:
xpack.canvas.expressionTypes => xpack.canvas.datasource for all ids in this file. Let us discuss it